pkg/node/node.go

```diff
@@ -16,3 +16,30 @@ func GetNodeName(node *v3.Node) string {
 	}
 	return ""
 }
+
+func GetEndpointNodeIP(node *v3.Node) string {
+	externalIP := ""
+	internalIP := ""
+	for _, ip := range node.Status.InternalNodeStatus.Addresses {
+		if ip.Type == "ExternalIP" && ip.Address != "" {
+			externalIP = ip.Address
+			break
+		} else if ip.Type == "InternalIP" && ip.Address != "" {
+			internalIP = ip.Address
+		}
+	}
+
+	if externalIP != "" {
+		return externalIP
+	}
+	return internalIP
+}
+
+func GetNodeByNodeName(nodes []*v3.Node, nodeName string) *v3.Node {
+	for _, m := range nodes {
```

Comment:
be careful when it comes to nodeName. For some nodes, machine.name in rancher does not equal to node name in k8s. You can look for the logic of looking up node by name here: https://github.com/rancher/rancher/pull/12936

the PR moves the method to helper so it can be used from other packages